tests/Analyzers/AnalyzerAssemblyGenerator.cs

```diff
@@ -30,9 +37,19 @@ private static async Task<IEnumerable<MetadataReference>> GetReferencesAsync()
                 MetadataReference.CreateFromFile(typeof(CodeFixProvider).Assembly.Location),
             };
 
-            var netcoreMetadataReferences = await ReferenceAssemblies.Net.Net70.ResolveAsync(LanguageNames.CSharp, CancellationToken.None);
+            // Resolve the targeting pack and the target framework used to compile the test assembly.
+            var netCurrentTargetingPackVersion = (string)AppContext.GetData("ReferenceAssemblies.NetCurrent.TargetingPackVersion")!;
+            var netCurrentTargetFramework = (string)AppContext.GetData("ReferenceAssemblies.NetCurrent.TargetFramework")!;
+            var nugetConfigPath = (string)AppContext.GetData("ReferenceAssemblies.NetCurrent.NuGetConfigPath")!;
+            ReferenceAssemblies netCurrentReferenceAssemblies = new(netCurrentTargetFramework,
+                new PackageIdentity("Microsoft.NETCore.App.Ref", netCurrentTargetingPackVersion),
+                Path.Combine("ref", netCurrentTargetFramework));
+            netCurrentReferenceAssemblies = netCurrentReferenceAssemblies.WithNuGetConfigFilePath(nugetConfigPath);
+
+            var netcoreMetadataReferences = await netCurrentReferenceAssemblies.ResolveAsync(LanguageNames.CSharp, CancellationToken.None);
             references.AddRange(netcoreMetadataReferences.Where(reference => Path.GetFileName(reference.Display) != "System.Collections.Immutable.dll"));
```

Comment:
That code is still valid as System.Collections.Immutable.dll is part of the Microsoft.NETCore.App.Ref targeting pack. I have no context here but I assume the assembly is filtered out as it's coming in via an external dependency and this is a lazy attempt to replicate msbuild's conflict resolution.

This should probably be removed together with your above feedback
> All of these should be removed and resolved from NuGet packages instead of from assemblies on disk. The only time MetadataReference.CreateFromFile(typeof(T).Assembly.Location) is valid is when the test assembly is defined in the same repository as T.